### PR TITLE
Updated and modernized docs and CLI

### PR DESCRIPTION
**Pull request by sg-s**
## changes

- overhauled docs, now points to low-level generated code
- numerous small bugfixes and typo fixes

### CODE REVIEW

File: docs/how-to/managed-data/upload-data.md [hunk 1]
```diff
@@ -85,12 +86,12 @@ This page describes how to upload data to Deep Origin Data.
...
 
     ```py
-    from deeporigin.managed_data import _api, api
-    _api.upload_file("/path/to/file.fasta")
+    from deeporigin.managed_data import api
+    api.upload_file("/path/to/file.fasta")
     ```
 
     This uploads files to the "Staging Area" of Deep Origin
-    Data, and returns data that conforms to [DescribeFileResponse][src.managed_data.schema.DescribeFileResponse]. An example is shown below:
```
Comment (sg-s) [inline docs/how-to/managed-data/upload-data.md:93]: unfortunately these models have gone away and i haven't found an easy way to include the models in the generated code back here

File: docs/ref/managed-data/high-level-api.md [hunk 2]
```diff
@@ -5,7 +5,10 @@
 
 ::: src.managed_data.api
     options:
-      filters: ["!^_[^_]"]
+      filters: 
+        - "!^_"
+        - "!ensure_client"
+        - "!add_row_to_data"
```
Comment (sg-s) [inline docs/ref/managed-data/high-level-api.md:11]: suppressing some methods that shouldn't be shown to users

File: tests/mock_client.py [hunk 3]
```diff
@@ -335,6 +335,29 @@ def create_file_download_url(self, **kwargs):
...
+            uri="s3://data.deeporigin-com.ijvjf/files/placeholder",
+            contentType=content_type,
+            dateCreated="2024-07-25T15:24:09.459Z",
+        )
+
+        return types.create_file_upload_response.Data(
+            file=file,
+            uploadUrl="https://foo.bar/foo",
+        )
+
```
Comment (sg-s) [inline tests/mock_client.py:360]: this allows us to test that function using a mock client

**Review by jonrkarr — APPROVED** [review]
I haven't reviewed the changes. Rather than review these changes, I plan to review all of the documentation. I want to focus on looking at the documentation from the user's perspective, rather than on changes.

**Pull request merged**